Addded TeamsAIPolicy Resource

TorbenA opened the pull request:
#### Pull Request (PR) description
Added TeamsAIPolicy Resource


#### Task list

- [x] Added an entry to the change log under the Unreleased section of the file CHANGELOG.md.
      Entry should say what was changed and how that affects users (if applicable), and
      reference the issue being resolved (if applicable).
- [x] Resource parameter descriptions added/updated in the schema.mof.
- [x] Resource documentation added/updated in README.md.
- [x] Resource settings.json file contains all required permissions.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated.
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

TorbenA committed:
Added MSFT_TeamsAIPolicy

TorbenA committed:
Added example and tests for TeamsAIPolicy

TorbenA committed:
Updated Changelog for TeamsAIPolicy

TorbenA committed:
Added Stubs for TeamsAIPolicy

FabienTschanz commented on Modules/Microsoft365DSC/DSCResources/MSFT_TeamsAIPolicy/MSFT_TeamsAIPolicy.psm1:
```diff
@@ -0,0 +1,411 @@
+            return $nullResult
+        }
+
+        Write-Verbose -Message "Found an instance with Identity {$Identity}"
```
Please improve the verbose statement by adding more context about the resource. This isn't (yet) consistent across all of the existing resources, but we do want that for all of them at the end.
```suggestion
        Write-Verbose -Message "A Teams AI Policy with Identity {$Identity} was found"
```

TorbenA replied:
Changed in https://github.com/microsoft/Microsoft365DSC/pull/6655/commits/ca6f7679952f66db8ea25c2e6940354340cec6db

FabienTschanz commented on Modules/Microsoft365DSC/DSCResources/MSFT_TeamsAIPolicy/MSFT_TeamsAIPolicy.psm1:
```diff
@@ -0,0 +1,411 @@
+    [CmdletBinding()]
+    [OutputType([System.String])]
+    param
+    (
```
Please add the `$Filter` parameter as well for filtering purposes. You can get inspiration from other resources if you like.

TorbenA replied:
Added in https://github.com/microsoft/Microsoft365DSC/pull/6655/commits/ca6f7679952f66db8ea25c2e6940354340cec6db
Since the Teams PowerShell (in contrast to the Graph PowerShell) does not accept $null or "" as -Filter parameter I needed to add "*" to the parameter. Since this seems to be the first Teams resource to support the $Filter parameter please let me know if this is an acceptable solution.

FabienTschanz replied:
If it works like that, then absolutely, yes 👍

FabienTschanz commented on Modules/Microsoft365DSC/DSCResources/MSFT_TeamsAIPolicy/MSFT_TeamsAIPolicy.schema.mof:
```diff
@@ -0,0 +1,16 @@
+    [Write, Description("Provides explanatory text about the Teams AI policy.")] String Description;
+    [Write, Description("EnrollFace controls user access to user face enrollment in the Teams app settings.")] String EnrollFace;
+    [Write, Description("EnrollVoice controls user access to user voice enrollment in the Teams app settings.")] String EnrollVoice;
+    [Write, Description("Setting to Enabled turns on speaker attribution in BYOD scenarios while Disabled will turn off the function.")] String SpeakerAttributionForBYOD;
```
Please add the `ValueMap{"Enabled","Disabled"}, Values{"Enabled","Disabled"}` for value mapping. See the `Ensure` parameter further below.

TorbenA replied:
Added in https://github.com/microsoft/Microsoft365DSC/pull/6655/commits/ca6f7679952f66db8ea25c2e6940354340cec6db

FabienTschanz reviewed (changes requested):
Looks great so far, thank you very much. I added some comments. Please address them and add an example for Update and Remove as well. After that, we can check it again and merge.

TorbenA committed:
Added $Filter parameter to Teams AI Policy
Added ValidateSets to Schema for Teams AI Policy
Added Examples for Update and Remote to Teams AI Policy
